File: java/src/main/java/com/svix/agentic_commerce_protocol/ErrorType.java
// This file is @generated
package com.svix.agentic_commerce_protocol;

import com.fasterxml.jackson.annotation.JsonValue;
import com.svix.agentic_commerce_protocol.Utils.ToQueryParam;

public enum ErrorType implements ToQueryParam {
    INVALID_REQUEST("invalid_request"),
    REQUEST_NOT_IDEMPOTENT("request_not_idempotent"),
    PROCESSING_ERROR("processing_error"),
    SERVICE_UNAVAILABLE("service_unavailable");
    private final String value;

    ErrorType(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }

    @Override
    public String toQueryParam() {
        return this.value;
    }
}

File: kotlin/src/main/kotlin/com/svix/agentic_commerce_protocol/ErrorType.kt
// This file is @generated
package com.svix.agentic_commerce_protocol

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.encodeToJsonElement
import kotlinx.serialization.json.jsonPrimitive

@Serializable
enum class ErrorType : ToQueryParam {
    @SerialName("invalid_request") INVALID_REQUEST,
    @SerialName("request_not_idempotent") REQUEST_NOT_IDEMPOTENT,
    @SerialName("processing_error") PROCESSING_ERROR,
    @SerialName("service_unavailable") SERVICE_UNAVAILABLE;

    override fun toQueryParam() = Json.encodeToJsonElement(this).jsonPrimitive.content
}
